gc/base/MemorySubSpace.cpp

```diff
@@ -1929,3 +1936,28 @@ MM_MemorySubSpace::wasContractedThisGC(uintptr_t gcCount)
 	return (gcCount == _extensions->heap->getResizeStats()->getLastHeapContractionGCCount());
 }
 
+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+uintptr_t
+MM_MemorySubSpace::releaseFreeMemoryPages(MM_EnvironmentBase* env)
+{
+	Assert_MM_unreachable();
+        return 0;
+}
+
+bool
+MM_MemorySubSpace::isSystemGCAllowed(MM_EnvironmentBase* env, uint32_t gcCode)
+{
+	bool invokeGC = true;
+	if (J9MMCONSTANT_IMPLICIT_GC_IDLE_GC == gcCode) {
+		MM_MemorySpace* defaultMemorySpace = _extensions->heap->getDefaultMemorySpace();
+		uintptr_t freeMemorySize = defaultMemorySpace->getApproximateActiveFreeMemorySize(MEMORY_TYPE_OLD);
+		uintptr_t actvMemorySize = defaultMemorySpace->getActiveMemorySize(MEMORY_TYPE_OLD);
+		uintptr_t previousMemorySize = 0 < _extensions->lastGCFreeBytes ? _extensions->lastGCFreeBytes : actvMemorySize;
+
+		if ((0 < freeMemorySize) && (20 > (((previousMemorySize - freeMemorySize) * 100) / actvMemorySize))) {
```

Comment:
The threshould should be a configurable parameter, via command line option. 

The logic is not right to me - I believe we discussed this. It should be relative to historic free memory, not relative to active. The logic is: if we only have <60 or so % of free memory (currently) available relative to what we historically have had, just after a global GC, than we should do an explicit sysGC to regain even those extra 40%.

The current criteria does not guarantee that any new free memory will be made available. For example, if we historically free up 30 and currently we are also 30 you'll get (30 - 30) < 20, and sysGC will be triggered, but actually 0 additional memory will be freed.

The current criteria has its own benefits, though. In an opposite example, if we already have very high hisstorical free memory, like 80&, and we are currently at 59, it will prevent a sysGC. 

A compromise is perhaps, to keep both...

I also suggest that you maintain historical weighted average of free memory. For example, avg = (50% of current + 50% of prev average) 
.